### PR TITLE
add kornia augmentations

### PR DESCRIPTION
**Pull request by edgarriba**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  

## What does this PR do?
Fixes # (issue).

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: kaggle_plantpatho/models.py [hunk 1]
```diff
@@ -52,6 +57,7 @@ def compute_loss(self, y_hat, y):
 
     def training_step(self, batch, batch_idx):
         x, y = batch
+        x = self.aug(x)  # => batched augmentations
```
Comment (Borda) [inline kaggle_plantpatho/models.py:60]:
```suggestion
        if self.aug:
            x = self.aug(x)  # => batched augmentations
```

File: kaggle_plantpatho/models.py [hunk 2]
```diff
@@ -35,6 +39,7 @@ def __init__(self, model, lr: float = 1e-4):
         self.val_accuracy = torchmetrics.Accuracy()
         self.val_f1_score = torchmetrics.F1(self.num_classes, average='weighted')
         self.learn_rate = lr
+        self.aug = augmentations or LitAugmenter()
```
Comment (Borda) [inline kaggle_plantpatho/models.py:42]:
```suggestion
        self.aug = augmentations
```

**Review by Borda — COMMENTED** [review]
just to avoid multiple augmentations since there is one default for Dataset we shall not have another in the model...

File: kaggle_plantpatho/models.py [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
+
 import torch
 import torchmetrics
 import torchvision
 from pytorch_lightning import LightningModule
 from torch import nn
 from torch.nn import functional as F
 
+from kaggle_plantpatho.augment import LitAugmenter
+
```
Comment (Borda) [inline kaggle_plantpatho/models.py:11]:
```suggestion
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@edgarriba is the resize issue on Win because of using v0.4 instead of master?

**Comment by edgarriba** [comment]
Better master, we recently found a bug

File: requirements.txt [hunk 4]
```diff
@@ -1,8 +1,9 @@
 torch>=1.6
 torchmetrics>=0.2.0
-torchvision
 pytorch-lightning>=1.2.0
+torchvision
+kornia>=0.4
```
Comment (edgarriba) [inline requirements.txt:5]:
```suggestion
git+https://github.com/kornia/kornia
```

File: kaggle_plantpatho/augment.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        return geometry.resize(x[None], self.size)[0]
+
+
+class LitPreprocess(nn.Module):
+    """Applies the processing to the image in the worker before collate."""
+
+    def __init__(self, img_size: int):
+        super().__init__()
+        self.preprocess = nn.Sequential(
+            #K.augmentation.RandomResizedCrop((224, 224)),
```
Comment (edgarriba) [inline kaggle_plantpatho/augment.py:53]:
```suggestion
            # K.augmentation.RandomResizedCrop((224, 224)),
```

File: kaggle_plantpatho/augment.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        '''self.geometric = [
+            K.augmentation.RandomAffine(60., p=0.75),
+        ]'''
+        self.augmentations = nn.Sequential(
+            augmentation.RandomRotation(degrees=30.),
+            augmentation.RandomPerspective(distortion_scale=0.4),
+            augmentation.RandomResizedCrop((224, 224)),
+            augmentation.RandomHorizontalFlip(p=0.5),
+            augmentation.RandomVerticalFlip(p=0.5),
+            #K.augmentation.GaussianBlur((3, 3), (0.1, 2.0), p=1.0),
```
Comment (edgarriba) [inline kaggle_plantpatho/augment.py:81]:
```suggestion
            # K.augmentation.GaussianBlur((3, 3), (0.1, 2.0), p=1.0),
```

File: kaggle_plantpatho/augment.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+            #K.augmentation.ColorJitter(0.01, 0.01, 0.01, 0.01, p=0.25),
```
Comment (edgarriba) [inline kaggle_plantpatho/augment.py:82]:
```suggestion
            # K.augmentation.ColorJitter(0.01, 0.01, 0.01, 0.01, p=0.25),
```

File: kaggle_plantpatho/augment.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        )
+        self.denorm = augmentation.Denormalize(Tensor(DATASET_IMAGE_MEAN), Tensor(DATASET_IMAGE_STD))
+
+    @torch.no_grad()
+    def forward(self, x: Tensor) -> Tensor:
+        assert len(x.shape) == 4, x.shape
+        out = x
+        #idx = torch.randperm(len(self.geometric))[0]  # OneOf
```
Comment (edgarriba) [inline kaggle_plantpatho/augment.py:90]:
```suggestion
        # idx = torch.randperm(len(self.geometric))[0]  # OneOf
```

File: kaggle_plantpatho/augment.py [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        #out = self.geometric[idx](x)
```
Comment (edgarriba) [inline kaggle_plantpatho/augment.py:91]:
```suggestion
        # out = self.geometric[idx](x)
```

**Comment by edgarriba** [comment]
@Borda what's the error about install-pkg ?

File: requirements.txt (hunk 4 above)
Comment (Borda) [reply requirements.txt]: this does not add KOrnian in Gh dependencies..

File: requirements.txt [hunk 10]
```diff
@@ -1,8 +1,9 @@
 torch>=1.6
 torchmetrics>=0.2.0
-torchvision
 pytorch-lightning>=1.2.0
+torchvision
+git+https://github.com/kornia/kornia
```
Comment (Borda) [inline requirements.txt:5]:
```suggestion
kornia>=0.4  # todo: it shall be 0.4.x or 0.5 as there is bug and performance issue
https://github.com/kornia/kornia/archive/refs/heads/master.zip
```

**Pull request merged**